### PR TITLE
[Nested Tensor] Add xpu device in assertion for nested tensor creation

### PR DESCRIPTION
**Pull request by zejun-chen**
Add xpu device checking in nested tensor creation

### CODE REVIEW

**Comment by zejun-chen** [comment]
Hi, @gujinghui @jgong5 
I add the xpu device in the assertion of the nested tensor creation.

**Review by jgong5 — APPROVED** [review]

File: aten/src/ATen/NestedTensorImpl.cpp [hunk 1]
```diff
@@ -179,7 +179,7 @@ NestedTensorImpl::NestedTensorImpl(
       "in the near future.");
   auto storage_device = storage_.device();
   TORCH_INTERNAL_ASSERT(
-      storage_device.is_cpu() || storage_device.is_cuda() || storage_device.is_privateuseone(),
+      storage_device.is_cpu() || storage_device.is_cuda() || storage_device.is_xpu() || storage_device.is_privateuseone(),
       "NestedTensorImpl storage must be either CUDA, CPU or ", get_privateuse1_backend(), " but got ",
```
Comment (xunnanxu) [inline aten/src/ATen/NestedTensorImpl.cpp:183]: shall we update this assertion message as well?

**Review by xunnanxu — APPROVED** [review]

File: aten/src/ATen/NestedTensorImpl.cpp (hunk 1 above)
Comment (zejun-chen) [reply aten/src/ATen/NestedTensorImpl.cpp]: OK. Fix it. Thank you for comments. :)

**Pull request closed without merging**